ext/storage_sources/s3_store/s3_store.cxx

```diff
@@ -0,0 +1,245 @@
+            std::cout << "All buckets under my account:" << std::endl;
+            for (const std::string &bucket : buckets) {
+                std::cout << "  * " << bucket << std::endl;
+            }
+            std::cout << std::endl;
+        }
+
+        /* Have at least one bucket to use. */
+        if (!buckets.empty()) {
+            const Aws::String first_bucket = buckets.at(0);
+
+            /* List objects. */
+            std::vector<std::string> bucket_objects;
+            if (fs->conn->list_objects(first_bucket, bucket_objects)) {
+                std::cout << "Objects in bucket '" << first_bucket << "':" << std::endl;
+                if (!bucket_objects.empty()) {
+                    for (const auto &object : bucket_objects) {
+                        std::cout << "  * " << object << std::endl;
+                    }
+                } else {
+                    std::cout << "No objects in bucket." << std::endl;
+                }
+                std::cout << std::endl;
+            }
+
+            /* Put object. */
+            fs->conn->put_object(first_bucket, "WiredTiger.turtle", "WiredTiger.turtle");
+
+            /* List objects again. */
+            bucket_objects.clear();
+            if (fs->conn->list_objects(first_bucket, bucket_objects)) {
+                std::cout << "Objects in bucket '" << first_bucket << "':" << std::endl;
+                if (!bucket_objects.empty()) {
+                    for (const auto &object : bucket_objects) {
+                        std::cout << "  * " << object << std::endl;
+                    }
+                } else {
+                    std::cout << "No objects in bucket." << std::endl;
+                }
+                std::cout << std::endl;
+            }
+
+            /* Delete object. */
+            fs->conn->delete_object(first_bucket, "WiredTiger.turtle");
+
+            /* List objects again. */
+            bucket_objects.clear();
+            if (fs->conn->list_objects(first_bucket, bucket_objects)) {
+                std::cout << "Objects in bucket '" << first_bucket << "':" << std::endl;
+                if (!bucket_objects.empty()) {
+                    for (const auto &object : bucket_objects) {
+                        std::cout << "  * " << object << std::endl;
+                    }
+                } else {
+                    std::cout << "No objects in bucket." << std::endl;
+                }
+                std::cout << std::endl;
+            }
+        } else {
+            std::cout << "No buckets in AWS account." << std::endl;
+        }
+    }
+
+    *file_systemp = &fs->file_system;
+    return 0;
+}
+
+/*
+ * s3_fs_terminate --
+ *     Discard any resources on termination of the file system.
+ */
+static int
+s3_fs_terminate(WT_FILE_SYSTEM *file_system, WT_SESSION *session)
+{
+    S3_FILE_SYSTEM *s3_fs;
+
+    UNUSED(session); /* unused */
+
+    s3_fs = (S3_FILE_SYSTEM *)file_system;
+    delete (s3_fs->conn);
+    free(s3_fs);
```

Comment:
@sulabhM We must free this inside the function? It is weird.

Reply:
Yes, this function is assigned to the function pointer ```file_system.terminate```. That is the one that gets called to destroy the file system. So we need to free any associated memory in this function.